src/cirrus/lib/events.py

```diff
@@ -153,12 +154,21 @@ def announce(self: Self, event: WorkflowEvent) -> None:
     def claim_processing(
         self: Self,
         payload_id: str,
+        arn_base: str,
```

Comment:
Passing in the ARN base here then explicitly not using it in the return is a bit smelly to me. What if you simply took an ARN in here, and had a separate utility function that, given an ARN base, would return an ARN and execution_name? You'd call that function prior to this one.

Reply:
You're also going to need to handle the case where the item is already in the `CLAIMED` state, e.g., `process` timed out before the step function could be started and/or the state updated to `PROCESSING`. I think this means in `ProcessPayloads.get_states()` you need to change

```
        return {c["payload_id"]: StateEnum(c["state"]) for c in self.state_items}
```

to something like 

```
        return {
            c["payload_id"]: {
                "state": StateEnum(c["state"]),
                "last_exec_arn": c["executions"][-1],
            }
            for c in self.state_items}
```

All uses of the states returned would obviously need to access the item state via the `"state"` key, but the reason for this change is to allow you to handle the `CLAIMED` state  appropriately in `ProcessPayloads.process()`. That is, if you have a record in the `CLAIMED` state you then need to pass the last execution ARN into `ProcessPayload.__call__()` so it can be reused (and you skip re-claiming). In cases where the state is not `CLAIMED` you'd generate a new ARN to be passed in to `ProcessPayload.__call__()`.

Thus, I think my suggestion to split the ARN generation from `claim_processing` is so you can actually call the suggested function in `ProcessPayloads.process()` to get a value to pass in to `ProcessPayload.__call__()`...

Maybe an even cleaner way to do this is to rewrite my suggestion a different way, perhaps:

```
        return {id: attrs for id, attrs in self.get_process_attrs(self.state_items)}

    @staticmethod
    def get_process_attrs(state_items):
        for si in state_items:
            payload_id = si["payload_id"]
            state = StateEnum(si["state"])

            if state == StateEnum.CLAIMED:
                exec_arn = si["executions"][-1]
            else:
                exec_arn = gen_exec_arn(arn_base, payload_id)

            yield payload_id, {
                "state": state,
                "exec_arn": exec_arn,
            }
```

Reply:
I believe the revised deterministic execution naming is aligned with your suggestions here.